presto-common/src/main/java/com/facebook/presto/common/RuntimeStats.java

```diff
@@ -74,15 +75,18 @@ public RuntimeMetric getMetric(String name)
     }
 
     @JsonValue
-    public Map<String, RuntimeMetric> getMetrics()
+    public ConcurrentHashMap<String, RuntimeMetric> getMetrics()
```

Comment:
Returning a reference to the `metrics` map and allowing for users of this class to have the ability to modify the state of this class is scary. This allows for leakage of logic of concurrency control outside of this class and would make it difficult to track down all future use cases of misuse of this.

We should basically return an `ImmutableMap.copyOf(metrics)` here instead to protect the value from being modified outside the control of this class.

Reply:
I agree that returning a reference is not good. I have updated the callers to avoid updating this map outside of the class.

Using ImmutableMap.copyOf to make a copy will introduce some overhead which would be good to avoid. I kind of miss the c++ feature which can make a const reference without any overhead. Seems that there is no way to achieve that in Java.

Besides, this class is in the presto-common repo which doesn't depend on guava. There may be some concerns if we make this repo and the spi repo depend on guava, right?

Considering these two factors, should we keep this interface as-is?